docs/DESIGN.md

```diff
@@ -2,44 +2,146 @@
 
 ## Vision
 
-> High-level goals defining the long term vision for the project that will guide daily activities, its design, and feature acceptance.
+These are the high-level goals defining the long-term vision for the project that will guide daily activities, its design, and feature acceptance.
 
 - **High performance**: auto-instrumentation performance impact should not be a concern for its users.
 - **Reliability**: stable and performant under different loads. Well-behaved under extreme load, with predictable, low resource consumption.
 - **Visibility**: users should be able to generate telemetry data that provides deep and detailed visibility into their applications. Such telemetry must allow users to identify and solve application-related issues in production.
 - **Useful out-of-the-box**: after install users should be able to get telemetry data from targeted libraries with none or minimal configuration (good selection of defaults).
 - **Extensible**: key components can be chosen via configuration and plugins.
 
-## Constraints
-
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",
-"NOT RECOMMENDED", "MAY", and "OPTIONAL"  are to be interpreted as described in 
-[BCP 14](https://tools.ietf.org/html/bcp14), [[RFC2119](https://tools.ietf.org/html/rfc2119)], [[RFC8174](https://tools.ietf.org/html/rfc8174)]
-when, and only when, they appear in all capitals, as shown here.
-
-- MUST NOT reference external libraries to prevent instrumented applications from breaking or behaving unexpectedly.
-- Compliance to [Common Language Specification (CLS)](https://docs.microsoft.com/en-us/dotnet/standard/language-independence-and-language-independent-components)
-is OPTIONAL. More: [#131](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/131).
+## Architecture
+
+To instrument a .NET application without requiring source code changes, the OpenTelemetry .NET Instrumentation uses the
+[CLR Profiler APIs](https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/)
+to bootstrap the [OpenTelemetry .NET SDK](https://github.com/open-telemetry/opentelemetry-dotnet#readme)
+and inject the selected instrumentations into the targeted application.
+
+The main components of the project are:
+
+- [**CLR Profiler Dll**](../src/OpenTelemetry.AutoInstrumentation.ClrProfiler.Native):
+a native component that implements a CLR Profiler. The CLR Profiler is used to
```

Comment:
All of the above :) if I had to give lower priority to something it would be mac and arm64. Mac because most of apps interested in auto-instrumentation are servers and typically not run on Mac. arm64 just because is more recent and has smaller adoption. That said we should target all of the above.